### ISSUE TITLE
Inserting children before or after JSXElement failing

### ISSUE THREAD

**Issue by AbhiSivaprasad** [babel/babel#9682]
## Bug Report

**Input Code**
```js
<div>
  <span>text</span>
</div>

spanPath.insertAfter(JSXExpressionContainerNode)
```

**Expected behavior/code**

```js
<div>
  <span>text</span>
  { /* JSXExpressionContainer */}
</div>
```

**Current Behavior**
Currently it is possible to insert a JSX Element as a sibling to a JSX element child using path.insertAfter or path.insertBefore. However, inserting a JSX expression container as a sibling is failing even though it is legal to have a JSX expression container sibling in the AST.

```
TypeError: Property body[0] of BlockStatement expected node to be of a type ["Statement"] but instead got "JSXExpressionContainer"
at validate (node_modules/@babel/types/lib/definitions/utils.js:128:13)
     at validator (node_modules/@babel/types/lib/definitions/utils.js:97:7)
     at Object.validate (node_modules/@babel/types/lib/definitions/utils.js:172:7)
     at validate (node_modules/@babel/types/lib/validators/validate.js:17:9)
     at builder (node_modules/@babel/types/lib/builders/builder.js:46:27)
     at Object.BlockStatement (node_modules/@babel/types/lib/builders/generated/index.js:260:31)
     at NodePath.replaceExpressionWithStatements (node_modules/@babel/traverse/lib/path/replacement.js:209:57)
     at NodePath.insertAfter (node_modules/@babel/traverse/lib/path/modification.js:128:17)
     at NodePath.replaceWithMultiple (node_modules/@babel/traverse/lib/path/replacement.js:85:22)
     at PluginPass.JSXElement (src/codemod/javascript/jsx.ts:30:17)
```

Scanning @babel/types, I see that JSXExpressionContainer does not alias with expression or statement which insertAfter and insertBefore require. However, since this is a valid AST it should be possible.

**Comment by tanhauhau** [comment]
Hi @AbhiSivaprasad, 
I couldn't reproduce this case, see [AST explorer](https://astexplorer.net/#/gist/27fed1c81451a3539211c67f78d44fad/52e4f880c25813d79eb8be857174a2641c6fc577)

Most likely your case has been fixed by https://github.com/babel/babel/issues/6464

**Comment by nicolo-ribaudo** [comment]
What does `npx nls why @babel/traverse` say? As @tanhauhau pointed out, it should have been fixed in v7.1.3
